frontend/src/components/AccountDetailsEditContainer.vue

```diff
@@ -0,0 +1,225 @@
+            </td>
+          </tr>
+        </table>
+        <div class="flex items-center pt-4">
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="reset()"
+          >
+            Abbrechen
+          </ButtonComponent>
+          <div class="flex-grow text-center">
+            <div class="text-red-500 italic" v-if="displayError">
+              Ungültige Eingabe
+            </div>
+            <div class="text-green-500 italic" v-if="displaySuccess">
+              Account gespeichert
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="updateInformation()"
+          >
+            Speichern
+          </ButtonComponent>
+        </div>
+        <div :class="showadminview ? 'flex items-center pt-1' : 'collapse'">
+          <div class="flex-grow text-center">
+            <div class="text-green-500 italic" v-if="deleted">
+              Account gelöscht
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="deleteAccount()"
+          >
+            Löschen
+          </ButtonComponent>
+        </div>
+      </template>
+    </ContainerComponent>
+  </div>
+</template>
+<script setup lang="ts">
+import { ref } from 'vue';
+import TextInputComponent from './TextInputComponent.vue';
+import ButtonComponent from './ButtonComponent.vue';
+import ContainerComponent from './ContainerComponent.vue';
+import { IUser, UserClass } from '../../../model/user/IUser';
+import { AccountRestInterface } from '../restInterfaces/AccountRestInterface';
+
+const props = defineProps({
+  /**
+   * Whether the delete button is displayed
+   */
+  showadminview: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  /**
+   * The user to display
+   */
+  user: {
+    type: Object as () => Partial<IUser>,
+    required: true,
+  },
+});
+
+const nameField = ref(props.user.name);
+const mailField = ref(props.user.mail);
+const pwField = ref('');
+const pwFieldControl = ref('');
+const disableInputShowSpinner = ref(false);
+const deleted = ref(false);
+const displayError = ref(false);
+const displaySuccess = ref(false);
+let originalUser = props.user as IUser;
+const isAdmin = ref(originalUser.type === UserClass.ADMIN);
+/**
+ * toggles the checkbox
+ */
+function toggleCheckBox(): void {
+  if (!disableInputShowSpinner.value) {
+    isAdmin.value = !isAdmin.value;
+  }
+}
+
+/**
+ * Updates Information from the User with the entered information
+ */
+async function updateInformation(): Promise<void> {
+  if (deleted.value) {
+    return;
+  }
+  displaySuccess.value = false;
+  displayError.value = false;
+  disableInputShowSpinner.value = true;
+
+  const updateUser: Partial<IUser> = { id: originalUser.id };
+
+  if (nameField.value !== '') {
```

Comment:
I dont think we should check for them to be empty in this way. The user should have the choice to do so. It should display an error when trying to save then

Reply:
hm since we dont enforce the user to set a new password when changing the mail or username (or do we want to?), it would be the consistent way to handle empty fields

Reply:
But since we're pre-filling, we'd have to check them against the original user, not empty